client_integration_test.go

```diff
@@ -651,6 +651,50 @@ func TestClientRemove(t *testing.T) {
 	}
 }
 
+func TestRemoveAll(t *testing.T) {
+	sftp, cmd := testClient(t, READWRITE, NODELAY)
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	// Create a temporary directory for testing
+	tempDir, err := ioutil.TempDir("", "sftptest-removeAll")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Create file and directory within the temporary directory
+	f, err := ioutil.TempFile(tempDir, "sftptest-removeAll*.txt")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+
+	d, err := ioutil.TempDir(tempDir, "sftptest-removeAll1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(d)
```

Comment:
Is this necessary, when it is a child of `tempDir` which means it should be deleted in line 664?